### PR TITLE
v1

### PR DESCRIPTION
**Pull request by dawnerd**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ],
+  userAgentRules: [
+    'mj12bot' // Generates excessive 404s
+  ]
+};
+
+const register = function(server, options) {
+  const settings = Object.assign({}, defaults, options);
+
+  server.ext('onPreResponse', (request, h) => {
```
Comment (ecwillis) [inline index.js]: can we make this an earlier lifecycle point? At this point in the request the entire handler has already been run.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const userAgent = request.headers['user-agent'].toLowerCase();
+    const acceptsHtml = (request.headers.accept || '').includes('html');
+    let matched = false;
+
+    if (acceptsHtml) {
+      return h.continue;
```
Comment (ecwillis) [inline index.js:29]: Is this going to break any API routes (in a general sense)... Or is this lib intended to only go on say "web" packages?

**Review by ecwillis — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (dawnerd) [reply index.js:29]: It shouldn't unless the API tries to hit one of the routes in the rules or sets a bad UA.

File: index.js (hunk 1 above)
Comment (dawnerd) [reply index.js]: So I tried with onPreHandler but it wasn't triggering if the route didn't exist (kinda the whole point). I'll dig in and see if there's something else

**Review by ecwillis — APPROVED** [review]

**Pull request merged**